src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java

```diff
@@ -64,6 +67,38 @@ public void whenTryingToSaveBookWithoutAuthorExpectNoSave() {
         Assertions.assertEquals(0, bookService.count());
     }
 
+    /**
+     * Tests book is not saved with numberOfPages > 23000
+     */
+    @Test
+    public void whenTryingToSaveBookWithMaxNumberPageExceedNoSave() {
+        Book book = new Book("Book without author", new Author("First", "Last"), toRead);
+        book.setNumberOfPages(Integer.MAX_VALUE);
+        try {
+            bookService.save(book);
+        } catch (Exception e)
+        {
+            Assertions.assertTrue(ExceptionUtils.getRootCause(e) instanceof ConstraintViolationException);
+        }
+        Assertions.assertEquals(0, bookService.count());
+    }
+
+    /**
+     * Tests book is not saved with pagesRead > 23000
+     */
+    @Test
+    public void whenTryingToSaveBookWithPagesReadExceedNoSave() {
+        Book book = new Book("Book without author", new Author("First", "Last"), toRead);
+        book.setPagesRead(Integer.MAX_VALUE);
```

Comment:
Instead of testing `Integer.MAX_VALUE`, could you test the one number above the boundary (23,001)?

Also, could you add tests for the boundary and one number less than the boundary?